src/main/java/net/dv8tion/jda/core/events/message/MessageReceivedEvent.java

```diff
@@ -35,38 +37,109 @@ public MessageReceivedEvent(JDA api, long responseNumber, Message message)
         this.message = message;
     }
 
+    /**
+     * Returns the received {@link net.dv8tion.jda.core.entities.Message Message} object.
+     * <br>The Message will be of type {@link net.dv8tion.jda.core.entities.impl.ReceivedMessage ReceivedMessage}.
```

Comment:
Considering the impl package is not included in the documentation we cannot link to it here

Reply:
Oh yea... Should i just remove that completely?